### PR TITLE
Eureka codegeneration

### PR DESCRIPTION
**Pull request by pwizenty**

### CODE REVIEW

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/pom.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>de.fhdo.lemma.model_processing.code_generation.springcloud.eureka</groupId>
+    <artifactId>de.fhdo.lemma.model_processing.code_generation.springcloud.eureka</artifactId>
+    <version>0.7.0-SNAPSHOT</version>
+    <name>Eureka Code Generator</name>
+    <description>A LEMMA code generator for eureka components.</description>
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/pom.xml:9]: eureka -> Eureka

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    /**
+     * Create container specific Eureka configuration for each deployed Spring-based microservice.
+     */
+    private def createContainerSpecificConfiguration(IntermediateContainer container) {
+        container.deployedServices.forEach[createServiceSpecificConfiguration(it)]
+    }
+
+    /**
+     * Create the Eureka service-specific configuration in the application.propertiers file.
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:88]: propertiers -> properties

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+     */
+    private def createServiceSpecificConfiguration(OperationMicroserviceReference service) {
+        val serviceFilePath = Util.buildPathFromQualifiedName(service.qualifiedName)
+
+        // FIXME: This statement only support Java or Kotlin microservices with the Spring Framework
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:93]: support -> supports

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+    /**
+     * Generate Eureka infrastructure node based on the spring cloud framework implemented in
+     * Kotlin.
+     */
+    private def createEurekaComponents(IntermediateInfrastructureNode node) {
+        // Generate pom.xml file
+        val pomFilePath = '''«targetFolder»«File.separator»«node.name.toLowerCase»''' +
+            '''«File.separator»pom.xml'''
+        this.content.put(pomFilePath, EurekaTemplate.generatePOMFile(node))
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:162]: Remove unnecessary "`this`es".

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+        // generate main class for the Spring-based Eureka implementation in Kotlin
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:164]: Again as for the Zuul generator: Please open an issue to also enable the generation of Java-based Eureka services in the future.

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    /**
+     * Build Eureka file path based on the service specific property groupId
+     */
+    private def String buildEurekaFilePath(IntermediateInfrastructureNode node) {
+        val property = node.defaultValues.findFirst[
+            it.technologySpecificProperty.name.toLowerCase == "mavengroupid"
+        ]
+        val groupIdPath = property.value.replaceAll("\\.", "/").toLowerCase
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:183]: cf. my comment on the Zuul generator regarding Java's `replaceAll`

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/intermediate_model_validation/EurekaIntermediateModelValidation.xtend [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import org.eclipse.xtext.validation.Check
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import de.fhdo.lemma.model_processing.code_generation.springcloud.eureka.code_generation.EurekaCodeGenerator
+import de.fhdo.lemma.operation.intermediate.IntermediateInfrastructureNode
+
+/**
+ * The Spring Cloud  Eureka model validator is responsible for checking the compatibility of the
```
Comment (frademacher) [inline code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/intermediate_model_validation/EurekaIntermediateModelValidation.xtend:12]: Remove duplicate whitespace

File: code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend (hunk 5 above)
Comment (pwizenty) [reply code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend:164]: Added the option to also generate a Java-based Eureka implementation.

**Pull request closed without merging**